From 3e497b852549c9199c4aa2eb34e6a989cb5fa2b8 Mon Sep 17 00:00:00 2001
From: "Brett T. Warden" <brett.t.warden@intel.com>
Date: Thu, 11 Apr 2024 16:43:07 -0700
Subject: [PATCH] Allow nut-scanner to match versioned libs

nut-scanner attempts to detect support libraries, like libusb-1.0,
libnetsnmp, etc., but only by matching the exact names with .so, for
example:
libusb-1.0.so
libnetsnmp.so

It will *not* match versioned shared objects, such as:
libusb-1.0.so.0
libusb-1.0.so.0.3.0
libnetsnmp.so.40
libnetsnmp.so.40.2.1

We only provide the bare .so files in devpkgs, so in most deployed
environments, nut-scanner will fail.

The code will still search for longer matches to *report*, so modify it to just use those.

The code isn't looking for actual libraries to load, it's just trying to
determine whether they exist. The actual UPS drivers will use dlopen to
load the correct libraries, so there should be little harm in allowing
the longer matches.
---
 common/common.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/common/common.c b/common/common.c
index fa6772165c32..4bce989cb427 100644
--- a/common/common.c
+++ b/common/common.c
@@ -2345,6 +2345,8 @@ static char * get_libname_in_dir(const char* base_libname, size_t base_libname_l
 				", but saw seemingly related names (are you missing"
 				" a symbolic link, perhaps?) e.g.: %s",
 				base_libname, dirname, libname_alias);
+			upsdebugx(1, "Clear Linux override -- using %s", libname_alias);
+			libname_path = xstrdup(libname_alias);
 		}
 
 		free(libname_alias);
-- 
2.44.0

